js/stanford_paragraph_types.p_hero.js

```diff
@@ -5,42 +5,132 @@
 
 (function ($) {
   Drupal.behaviors.stanfordParagraphPHero = {
+
     attach: function (context, settings) {
-      $('.field-name-field-p-hero-image', context).each(function () {
-        var video = $(this).siblings('.field-name-field-p-hero-video');
-
-        if (video.length) {
-          $(video).find('iframe').attr('title', Drupal.t('Video Player'));
-          var videoUrl = $(video).find('div[data-video-embed-url]').attr('data-video-embed-url');
-
-          var play = $('<a>', {
-            class: 'play-video',
-            href: videoUrl,
-            html: $('<i>', {
-              class: 'fa fa-youtube-play icon-youtube-play',
-              html: 'Play Video',
-              'aria-label': Drupal.t('Play Video - Opens to the video website')
-            })
-          }).click(function (e) {
-
-            // Mouse has eventPhase 3, keyboard has 2.
-            if (e.eventPhase == 3) {
-              e.preventDefault();
-              $dad = $(this).parent();
-              $dad.hide();
-              $dad.siblings('.group-overlay-text').hide();
-              var iframe = $(video).find('iframe')[0];
-
-              iframe.src += "&autoplay=1";
-              $(iframe).attr('onload', 'this.contentWindow.focus()');
-              $(video).show();
-            }
-
-          });
-
-          $(this).prepend(play);
+      var i = 0;
+      $('.field-name-field-p-hero-video iframe').each(function (i, iframe) {
+        // Make sure the video has an ID attribute.
+        if (typeof $(iframe).attr('id') == 'undefined') {
+          $(this).attr('id', 'hero-video-' + i);
+        }
+        i++;
+      });
+
+
+      var tag = document.createElement('script');
+      var firstScriptTag = document.getElementsByTagName('script')[0];
+
+      if ($('.field-name-field-p-hero-video iframe[src*="youtube"]').length) {
+        tag.src = '//www.youtube.com/iframe_api';
+        firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+      }
+
+      if ($('.field-name-field-p-hero-video iframe[src*="vimeo"]').length) {
+        $.getScript('//player.vimeo.com/api/player.js', function () {
```

Comment:
`//` is an anti pattern. This should always be https:// when it is available.

Reply:
is there a way to allow the js to load on `http` sites (local) too?

Reply:
i suppose i could check the current protocol huh?

Reply:
https resources will load on http hosted sites. Not the other way around. So when a remote resource has https available you should always use it.